src/sql/workbench/api/node/mainThreadNotebook.ts

```diff
@@ -32,9 +35,21 @@ export class MainThreadNotebook extends Disposable implements MainThreadNotebook
 		}
 	}
 
+	public addFuture(futureId: number, future: FutureWrapper): void {
```

Comment:
2 things need to be tracked in the main thread: what providers are registered, and what futures are registered. These are the only things with callbacks from the extension host. I chose not to use the same map for these just for simplicity, but can change that if requested.